### PR TITLE
Added bannedWordsAllowed attribute to Group attributes (Partial fix for #12405)

### PR DESCRIPTION
**Pull request by Scottd125**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Partial fix for #12405
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Added an optional Boolean attribute which will determine if banned words are allowed. Allows Step 1 to be completed and for Step 2 to become unblocked once merged and deployed to production. (As listed in the issue)

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 
9cbaf3a3-1f48-4557-9f0a-4d403df1f3eb

### CODE REVIEW

File: website/server/models/group.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ export const schema = new Schema({
     $type: String, enum: ['private', 'public'], default: 'private', required: true,
   },
   chat: Array, // Used for backward compatibility, but messages aren't stored here
+  bannedWordsAllowed: { $type: Boolean, required: false },
```
Comment (paglias) [inline website/server/models/group.js:88]: could you add this field to the list of un-settable fields here https://github.com/HabitRPG/habitica/pull/12440/files#diff-b6f239870392ed0ec2a2b73cbebaa699L153 (same file line 153) to make sure it cannot be set when the group is created or updated

**Review by paglias — CHANGES_REQUESTED** [review]
left one comment but otherwise it looks good!

File: website/server/models/group.js (hunk 1 above)
Comment (Scottd125) [reply website/server/models/group.js:88]: Sure thing, I can get this change done later today :)

**Comment by Scottd125** [comment]
I've got the requested change added in, give me a shout if there is anything else that needs to be done!

**Comment by paglias** [comment]
@Scottd125  thanks for this! I'm merging the PR and awarding your contributor tier.

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!

**Pull request merged**